veps/sig-compute/pcihole64-proposal.md

```diff
@@ -0,0 +1,323 @@
+`VirtualMachine` object to `"true"` the feature is enabled. Any other value
+than `"true"` will leave the feature disabled. By default, the feature will
+also be disabled.
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: my-legacy-vm
+  annotations:
+    kubevirt.io/legacyWindowsCompat: "true"
+spec:
+  runStrategy: Always
+  template:
+    spec:
+      domain:
+        devices: { }
+        memory:
+          guest: 512Mi
+      terminationGracePeriodSeconds: 180
+      volumes:
+        - containerDisk:
+            image: my.registry/images/legacy-os
+          name: legacy-os
+```
+
+## Alternatives
+
+<!--
+Outline any alternative designs that have been considered
+-->
+
+### Adding a specific field to the core API
+
+The feature could be configured by adding a specific field
+`spec.domain.devices.disablePCIHole64` of type `boolean` to the core API.
+
+Instead of enabling a broader compatibility behavior, this would express the
+configuration change to a `VirtualMachine` exactly.
+
+One advantage of this approach would be the improved discoverability of the
+feature, as it would become part of the core API and its documentation.
+
+The disadvantage of this approach, however, is that users must explicitly know
+that they need to disable the 64-bit PCI hole, and that the feature cannot be
+changed independently of the core API. Adding this field to the core API could
+also be interpreted as KubeVirt caring about OS specific configuration, which
+this proposal tries to avoid.
+
+### Extending KubeVirt with a hook plugin mechanism
+
+Another way to implement this feature is to extend KubeVirt with a plugin
+mechanism that would allow vendors to ship extensions to KubeVirt in a
+supportable manner, while keeping the changes required for this or other
+features outside the KubeVirt codebase.
+
+Today KubeVirt already has a feature called `Sidecar`s, which among other things
+allows to change the libvirt domain specification of a `VirtualMachine` before
+it is executed. A [proof of concept](#sidecar-proof-of-concept) for this
+proposal was created with the `Sidecar` feature. However, currently this
+feature is not supportable because, as can be seen in the PoC, it basically
+allows the execution of arbitrary code in the context of the `virt-launcher`
+pod.
+
+In order to make `Sidecar`s supportable, they would at least have to be changed
+so that only trusted code can be run as part of a hook. This could be realized
+for example by allowing to run only a single container image that contains all
+trusted hooks provided by the cluster admin. Before any hooks can be run, the
+cluster admin must deploy and configure this image in the cluster.
+
+The advantage of this approach would be that the KubeVirt codebase and API can
+be kept stable and clean, while vendors are still able to develop plugins for
+KubeVirt in a supportable way and with the least amount of effort required to
+maintain these plugins.
+
```

Comment:
This would also allow third parties to experiment with new APIs before introducing them to core KubeVirt.

Reply:
No. "Extending KubeVirt to allow using trusted or certified plugins trusted or certified plugins" is about allowing specific plugins, while the sidecar approach requires the cluster to allow any sidecar.

Reply:
Added a sentence mentioning it.